NonLocFEM/thermal_problems_2d.hpp

```diff
@@ -60,8 +65,8 @@ void solve_thermal_2d_problem(
         save_solution(solution, save, 0u);
     }
     for(const uint64_t step : std::ranges::iota_view{1u, time.steps_count + 1}) {
-        solver.calc_step(boundaries_conditions,
-            [right_part = auxiliary.right_part](const std::array<T, 2>& x) constexpr noexcept { return right_part; });
+        solver.calc_step(boundaries_conditions, 
+        [value = std::get<1>(auxiliary.right_part)](const std::array<T, 2>& x) constexpr noexcept { return value(x); });
```

Comment:
Аналогичное замечание
```suggestion
        [value = std::get<spatial_dependency<T, 2>>(auxiliary.right_part)](const std::array<T, 2>& x) constexpr noexcept { return value(x); });
```